types/rsync/index.d.ts

```diff
@@ -82,7 +82,10 @@ interface Rsync {
 interface RsyncStatic {
     new (): Rsync;
 
-    build: (options: Partial<{ [ Property in keyof Rsync ]: Parameters<Rsync[Property]>[0] }>) => Rsync;
+    // The `build` method will take an arguments object where the key is an rsync method and the value is the first paramater to that argument.
+    // If the method doesn't take an argument (e.g. `archive`, `progress`) then the value should be "true".
+    // Technically speaking, the value can be anything if the method doesn't take an argument, but it may as well be typed consistently here.
+    build: (options: Partial<{ [ Property in keyof Rsync ]: [Parameters<Rsync[Property]>[0]] extends [undefined] ? true : Parameters<Rsync[Property]>[0] }>) => Rsync;
```

Comment:
```suggestion
    build: (
    options: Partial<{
        [Property in keyof Rsync]: [Parameters<Rsync[Property]>[0]] extends [undefined]
            ? true
            : Parameters<Rsync[Property]>[0];
    }>,
) => Rsync;
```
this is after running DT prettier setup on the changed lines, just for better readability

Reply:
Thanks for the suggestion @peterblazejewicz I've added it to this PR